src/Tribe/Adjacent_Events.php

```diff
@@ -0,0 +1,324 @@
+<?php
+
+/**
+ * Controls getting a previous or next event from the context of a single event being viewed.
+ */
+class Tribe__Events__Adjacent_Events {
+
+	/**
+	 * @var int
+	 */
+	protected $current_event_id = 0;
+
+	/**
+	 * @var int
+	 */
+	public $previous_event_id;
+
+	/**
+	 * @var int
+	 */
+	public $next_event_id;
+
+	/**
+	 * @var string
+	 */
+	public $previous_event_link = '';
+
+	/**
+	 * @var string
+	 */
+	public $next_event_link = '';
+
+	/**
+	 * Set the "center" event ID to look on either side of in prev/next methods.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $event_id The event ID to look on either side of in prev/next methods.
+	 */
+	public function set_current_event_id( $event_id ) {
+		if ( 0 !== $this->current_event_id ) {
+			$this->current_event_id = $event_id;
```

Comment:
Why don't we allow the `current_event_id` to be set to 0? It's a good way to reset, no?

Reply:
I guess? I don't honestly understand why 0 would be acceptable, but will make the change to advance this PR.